interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -35,348 +38,444 @@ abstract class CatsEffectPlatform
+  private[this] val monadErrorInstance0: MonadError[Task, Throwable] =
+    new ZioMonadError[Any, Throwable]
 }
 
-private class CatsDefer[R, E] extends Defer[ZIO[R, E, *]] {
-  def defer[A](fa: => ZIO[R, E, A]): ZIO[R, E, A] = ZIO.effectSuspendTotal(fa)
+private class ZioDefer[R, E] extends Defer[ZIO[R, E, *]] {
+  type F[A] = ZIO[R, E, A]
+
+  override final def defer[A](fa: => F[A]): F[A] =
+    ZIO.effectSuspendTotal(fa)
 }
 
-private class CatsConcurrentEffect[R](rts: Runtime[R])
-    extends CatsConcurrent[R]
-    with effect.ConcurrentEffect[RIO[R, *]] {
-
-  override final def runAsync[A](fa: RIO[R, A])(
-    cb: Either[Throwable, A] => effect.IO[Unit]
-  ): effect.SyncIO[Unit] =
-    effect.SyncIO {
-      rts.unsafeRunAsync(fa.run) { exit =>
-        cb(exit.flatMap(identity).toEither).unsafeRunAsync(_ => ())
-      }
-    }
+private class ZioConcurrent[R, E] extends ZioMonadError[R, E] with GenConcurrent[ZIO[R, E, *], E] {
 
-  override final def runCancelable[A](fa: RIO[R, A])(
-    cb: Either[Throwable, A] => effect.IO[Unit]
-  ): effect.SyncIO[effect.CancelToken[RIO[R, *]]] =
-    effect.SyncIO {
-      rts.unsafeRun {
-        ZIO.descriptor
-          .bracketExit(
-            (descriptor, exit: Exit[Throwable, A]) =>
-              ZIO.effectTotal {
-                exit match {
-                  case Exit.Failure(cause) if !cause.interruptors.forall(_ == descriptor.id) => ()
-                  case _ =>
-                    effect.IO.suspend(cb(exit.toEither)).unsafeRunAsync(_ => ())
-                }
-              },
-            _ => fa
-          )
-          .interruptible
-          .forkDaemon
-          .map(_.interrupt.unit)
-      }
-    }
+  private def toPoll(restore: ZIO.InterruptStatusRestore) = new Poll[ZIO[R, E, *]] {
+    override def apply[T](fa: ZIO[R, E, T]): ZIO[R, E, T] = restore(fa)
+  }
+
+  private def toFiber[A](fiber: Fiber[E, A]) = new effect.Fiber[F, E, A] {
+    override final val cancel: F[Unit]           = fiber.interrupt.unit
+    override final val join: F[Outcome[F, E, A]] = fiber.await.map(toOutcome)
+  }
+
+  private def fiberFailure(error: E) =
+    FiberFailure(Cause.fail(error))
+
+  override def ref[A](a: A): F[effect.Ref[F, A]] =
+    ZRef.make(a).map(new ZioRef(_))
+
+  override def deferred[A]: F[Deferred[F, A]] =
+    Promise.make[E, A].map(new ZioDeferred(_))
 
-  override final def toIO[A](fa: RIO[R, A]): effect.IO[A] =
-    effect.ConcurrentEffect.toIOFromRunCancelable(fa)(this)
+  override final def start[A](fa: F[A]): F[effect.Fiber[F, E, A]] =
+    fa.interruptible.fork.map(toFiber)
+
+  override def never[A]: F[A] =
+    ZIO.never
+
+  override final def cede: F[Unit] =
+    ZIO.yieldNow
+
+  override final def forceR[A, B](fa: F[A])(fb: F[B]): F[B] =
+    fa.either *> fb
+
+  override final def uncancelable[A](body: Poll[F] => F[A]): F[A] =
+    ZIO.uninterruptibleMask(body.compose(toPoll))
+
+  override final val canceled: F[Unit] =
+    ZIO.interrupt.unit
```

Comment:
```suggestion
    ZIO.interrupt
```
Don't need `unit` here due to variance.